apps/piecemeal_app_creation_flow.go

```diff
@@ -0,0 +1,75 @@
+package apps
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	. "github.com/cloudfoundry/cf-acceptance-tests/cats_suite_helpers"
+
+	"github.com/cloudfoundry-incubator/cf-test-helpers/cf"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/helpers"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/app_helpers"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/assets"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/random_name"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = AppsDescribe("Piecemeal App Creation", func() {
+	var (
+		appName string
+	)
+
+	BeforeEach(func() {
+		appName = random_name.CATSRandomName("APP")
+	})
+
+	AfterEach(func() {
+		app_helpers.AppReport(appName)
+
+		Expect(cf.Cf("delete", appName, "-f", "-r").Wait()).To(Exit(0))
+	})
+
+	It("successfully creates and starts a new app with discrete commands", func() {
+		Expect(cf.Cf("create-app", appName).Wait()).To(Exit(0))
+		session := cf.Cf("app", appName, "--guid")
+		Eventually(session).Should(Exit(0))
+		appGUID := strings.TrimSpace(string(session.Buffer().Contents()))
+
+		Expect(cf.Cf("create-package", appName, "-p", assets.NewAssets().Dora).Wait()).To(Exit(0))
+
+		path := fmt.Sprintf("v3/apps/%s/packages", appGUID)
```

Comment:
Why not call `cf packages appName` instead?

Reply:
Confirmed our suspicion that the use of curl was for ease of response parsing.